### ISSUE TITLE
Cut+paste instead of merge when renaming folder to the same name as another folder

### ISSUE THREAD

**Issue by gerhaher** [koreader/koreader#4363]
* KOReader version: v2018.11.1-34-g3c6c15e_2018-11-21
* Device: KA1

#### Issue
If you have two folders, named A and B, and you (try to) rename the first to B, the folder will end up inside folder B, still named A.

In Windows in a similar situation the folders will be merged.

**Comment by cramoisi** [comment]
Windows will ask you if you want to merge :)

**Comment by pazos** [comment]
We're using common unix command 'mv' for moving/renaming, which in fact does what @gerhaher said. https://github.com/koreader/koreader/blob/2b9694d7d2c1a76e450e47a6e500cea9d871a9f3/frontend/apps/filemanager/filemanager.lua#L891-897

IMO this is the desired behaviour, but we can check if the destination is an existing folder and ask for merging (if source is a folder too). On bash something like:
```
if [ -d from ] && [ -d to ]; then
    mv from/* to/
fi
```

**Comment by mergen3107** [comment]
I think this was solver by #7833

**Comment by hius07** [comment]
No, it wasn't. Just checks for overwrite have been added. `mv` behaviour is unchanged.

**Comment by NiLuJe** [comment]
And we can't assume the platform's `mv` actually support merge workflows.

**Comment by mergen3107** [comment]
Ah, ok! Sorry for the confusion guys
